algorithms/preprocessing.py

```diff
@@ -1,7 +1,30 @@
 from supporting_components.graph import Graph, Vertex
+from algorithms.color_initialization import degree_color_initialization
+from algorithms.decide_gi import __is_balanced
 from math import factorial
 
 
+def could_be_isomorphic(G: 'Graph', H: 'Graph'):
+    """
+    Based on properties of the graphs a few simple comparisons are made to determine if the graphs could be isomorphic.
+    :param G, H: The graphs of which it should be determined if they could be isomorphic.
+    :return: Boolean that indicates if the two graphs could be isomorphic or not.
+    """
+    # Isomorphic graphs should have the same amount of vertices
+    if len(G.vertices) != len(H.vertices):
+        return False
+    # Isomorphic graphs should have the same amount of edges
+    if len(G.edges) != len(H.edges):
+        return False
+    # Isomorphic graphs should have the same amount of vertices with a specific degree
+    degree_color_initialization(G)
```

Comment:
Je gaat nu hier al kleuren op beide grafen, maar als ze isomorf kunnen zijn, kleur je ze later nog een keer. Kun je hier nog iets mee? Volgens mij hoef je dan op regel 21 en 34 van isomorphism_problem niet meer de degree_color_initialization te doen omdat beide grafen al gekleurd zijn.

Reply:
Ik heb hiervoor gekozen omdat je op basis van deze kleuring kan kijken of het aantal vertices met dezelfde degree hetzelfde is. Maar omdat we later (bijvoorbeeld bij het gebruik van twin removal) nog andere kleuringen kunnen gebruiken heb ik voor nu maar gekozen voor een eventuele dubbele kleuring.

Reply:
En nu ik de kleuring toepas op de disjoint_union, worden de originele grafen niet eens gekleurd. Dus er wordt nog steeds dubbel gekleurd, maar de originele grafen worden niet twee keer gekleurd.